### ISSUE TITLE
Keyboard Shortcuts: Improve Keyboard Navigation

### ISSUE THREAD

**Issue by BrittanyIRL** [GoogleForCreators/web-stories-wp#7912]
## Feature Description

Recently, while reviewing #7911, I noticed that the keyboard shortcuts dialog is actually not keyboard accessible. Work in #7478 has tended to the translation aspect of this dialog but adjusting the dialog to be keyboard friendly is out of scope. This ticket is for that work. 

The keyboard shortcut menu should be tabbable so that the labels of each keyboard shortcut get announced by a screen reader. The dialog title should be announced. And the close button should not act as a focus trap. 

On review of staging, this is  not a new behavior - think this has been this way for a while and this dialog has just been through some iterations and keyboard a11y got missed somewhere. 

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by BrittanyIRL** [comment]
Hey @csossi  if there's any way you can prioritize this one it would help unblock parts of our board. It should be easy to test: 

- Verify new toggle for keyboard shortcuts menu works the same as before (clicking once opens, clicking again closes. Same goes for 'enter')

**Comment by csossi** [comment]
Verified in QA
